src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -182,7 +186,33 @@ private Uni<Supplier<CommandResult>> executeCollectionCreation(QueryExecutor que
               } else {
                 return new SchemaChangeResult(true);
               }
-            });
+            })
+        .onFailure(
+            error ->
+                error instanceof InvalidQueryException
+                    && error
+                        .getMessage()
+                        .matches(
+                            ".*Cannot have more than \\d+ indexes, failed to create index on table.*"))
+        .recoverWithUni(
+            error ->
+                deleteCollectionOperation
```

Comment:
Are there possible side effects to reusing the delete collection operation here?

Reply:
It seems to me there shouldn't; operations being relatively simple containers of functionality?
(but asking @maheshrajamani to confirm).

Reply:
Drop collection sometimes produces timeouts.

Reply:
I mean, we have precedent, for example, deleteOperation will reuse findOperation?